dockerplugin.py

```diff
@@ -412,11 +465,88 @@ def read_callback(self):
                     if value:
                         method(container, cstats.dimensions, value, read_at)
             except Exception, e:
-                collectd.warning(('Error getting stats for container '
-                                  '{container}: {msg}')
-                                 .format(container=_c(container), msg=e))
-                collectd.warning(traceback.format_exc())
+                log.warning(('Unable to retrieve stats for container '
+                             '{container}: {msg}')
+                            .format(container=_c(container), msg=e))
+                # Print stack trace to log if verbose logging is on
+                log.info(traceback.format_exc())
+
+
+class CollectdLogHandler(logging.Handler):
+    """CollectdLogHandler
+    A custom log handler that forwards log messages raised
+    at level debug, info, notice, warning, and error
+    to collectd's built in logging.  Suppresses extraneous
+    info and debug statements using a "verbose" boolean
+    """
+    def __init__(self, plugin="unknown", verbose=False):
+        """@constructor
+        @param plugin : str
+        @param verbose : str
+        """
+        self.verbose = verbose
+        self.plugin = plugin
+        logging.Handler.__init__(self, level=logging.NOTSET)
+
+    def emit(self, record):
+        """
+        Emits a log record to the appropraite collectd log function
+        @param record : obj
+        """
+        try:
+            if record.levelname == 'ERROR':
+                if record.msg is not None:
```

Comment:
Don't repeat that if statement everywhere. Just check once at the top and return without doing anything if `record.msg is None`